x-pack/test/apm_api_integration/tests/alerts/wait_for_rule_status.ts

```diff
@@ -51,6 +56,40 @@ export async function waitForRuleStatus({
   });
 }
 
+export async function waitForDocumentInIndex<T>({
+  es,
+  indexName,
+  waitMillis = 10000,
+  log,
+}: {
+  es: Client;
+  indexName: string;
+  waitMillis?: number;
+  log: ToolingLog;
+}): Promise<SearchResponse<T, Record<string, AggregationsAggregate>>> {
+  if (waitMillis < 0) {
+    expect().fail(`waiting for ${indexName} timed out`);
+  }
+  try {
+    const response = await es.search<T>({ index: indexName });
+    if (response.hits.hits.length > 0) {
+      return response;
+    }
+  } catch (e) {
+    // Do nothing
+  }
+  log.debug(`Waiting for documents in index: ${indexName}, retrying...`);
+
+  await delay(WAIT_FOR_STATUS_INCREMENT);
+
+  return await waitForDocumentInIndex({
+    es,
+    indexName,
+    waitMillis: waitMillis - WAIT_FOR_STATUS_INCREMENT,
+    log,
+  });
+}
```

Comment:
`p-retry` will simplify your life (nothing less) 

```suggestion
async function waitForDocumentInIndex<T>({
  es,
  indexName,
}: {
  es: Client;
  indexName: string;
}): Promise<SearchResponse<T, Record<string, AggregationsAggregate>>> {
  return pRetry(async () => {
    const response = await es.search<T>({ index: indexName });
    if (response.hits.hits.length === 0) {
      throw new Error("No hits found");
    }
    return response;
  });
}
```

Reply:
nice, thanks for that. I updated both functions.